docs/spec_draft.md

```diff
@@ -550,6 +551,64 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-documented-ops)
 
+## stablehlo.iota
+
+### Semantics
+Fills a `result` array with values in increasing order starting from zero along
+the `iota_dimension` dimension.
```

Comment:
As discussed in #248, let's define op semantics from first principles, using only the definitions in this spec and college maths. This sentence is fine for building intuition for what's happening, but we'll also need to describe what  it means specifically, e.g. similarly to what https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#stablehlotranspose does.

Reply:
More specifically, here's how this could look like using Sandeep's suggestion from above and bringing it more in line with `concatenate`, `reshape` and `transpose`:

```
Fills a `result` tensor with values in increasing order starting from zero along
the `iota_dimension` dimension. More formally, 
`result[i0, ..., id, ..., iR-1] = id`, where `d` is equal to `iota_dimension`.
```